crates/migrate/src/lib.rs

```diff
@@ -0,0 +1,687 @@
+            false AS "is_default!: bool",
+            hmac_keys,
+            broker_address,
+            reactor_address,
+            ops_logs_name AS "ops_logs_name: models::Collection",
+            ops_stats_name AS "ops_stats_name: models::Collection"
+        FROM data_planes
+        WHERE data_plane_name = $1
+        "#,
+        name
+    )
+    .fetch_one(pg_pool)
+    .await
+    .with_context(|| format!("failed to fetch data-plane {name}"))?;
+
+    // Resolve ops collection templates for this data-plane.
+    let r = sqlx::query!(
+        r#"
+        SELECT
+            l.built_spec->'partitionTemplate' AS "logs!:  sqlx::types::Json<broker::JournalSpec>",
+            s.built_spec->'partitionTemplate' AS "stats!: sqlx::types::Json<broker::JournalSpec>"
+        FROM live_specs l, live_specs s
+        WHERE l.catalog_name = $1 AND l.spec_type = 'collection'
+        AND   s.catalog_name = $2 AND s.spec_type = 'collection'
+        "#,
+        &row.ops_logs_name,
+        &row.ops_stats_name,
+    )
+    .fetch_one(pg_pool)
+    .await
+    .context("failed to fetch data-plane ops collections")?;
+    let (ops_logs_template, ops_stats_template) = (r.logs.0, r.stats.0);
+
+    let mut metadata = gazette::Metadata::default();
+    metadata
+        .signed_claims(
+            proto_gazette::capability::APPEND
+                | proto_gazette::capability::APPLY
+                | proto_gazette::capability::LIST
+                | proto_gazette::capability::READ,
+            &row.data_plane_fqdn,
+            std::time::Duration::from_secs(300),
+            &row.hmac_keys,
+            broker::LabelSelector::default(),
+            "migrate-tool",
+        )
+        .context("failed to sign claims for data-plane")?;
+
+    // Create the journal and shard clients that are used for interacting with the data plane
+    let router = gazette::Router::new("local");
+    let journal_client =
+        gazette::journal::Client::new(row.broker_address.clone(), metadata.clone(), router.clone());
+    let shard_client = gazette::shard::Client::new(row.reactor_address.clone(), metadata, router);
+
+    Ok(DataPlane {
+        row,
+        shard_client,
+        journal_client,
+        ops_logs_template,
+        ops_stats_template,
+    })
+}
+
+struct SpecMigration {
+    catalog_name: String,
+    live_spec_id: models::Id,
+    built_spec: proto_flow::AnyBuiltSpec,
+}
+
+fn fetch_spec_migrations<'a>(
+    pg_pool: &'a sqlx::PgPool,
+    catalog_prefix: &'a str,
+    data_plane_id: models::Id,
+) -> impl futures::Stream<Item = anyhow::Result<SpecMigration>> + 'a {
+    sqlx::query!(
+        r#"
+        SELECT
+            id AS "id: models::Id",
+            catalog_name,
+            spec_type AS "spec_type!: models::CatalogType",
+            built_spec AS "built_spec!: sqlx::types::Json<models::RawValue>"
+        FROM live_specs
+        WHERE starts_with(catalog_name, $1)
+        AND   spec_type IS NOT NULL
```

Comment:
We don't actually set `spec_type` to null when deleting specs anymore, only the `spec` and `built_spec`. So I'm thinking this should change to `spec is not null`, assuming the desire is to filter out deleted specs.

Reply:
:+1: updated